[CI] MixedClusterEsqlSpecIT test {date_nanos.Date_nanos to date nanos, index version SYNC} failing

**Build Scans:**
- [elasticsearch-intake #10228 / 9.0.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qwcmqt3diqnpy)
- [elasticsearch-pull-request #33088 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/tqnd3v3vieobe)
- [elasticsearch-pull-request #32752 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/arp4wnne7etry)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v9.0.0#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {date_nanos.Date_nanos to date nanos, index version SYNC}" -Dtests.seed=2DE605DD9B4797AE -Dtests.bwc=true -Dtests.locale=de -Dtests.timezone=Europe/Paris -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bdate_nanos.Date_nanos%20to%20date%20nanos%2C%20index%20version%20SYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 0 column 0:a list containing
row 0 column 0:0: expected "2023-10-23T13:55:01.543123456Z" but was "2023-10-23T12:27:28.948Z"
row 1 column 0:a list containing
row 1 column 0:0: expected "2023-10-23T13:53:55.832987654Z" but was "2023-10-23T12:15:03.360103847Z"
row 2 column 0:a list containing
row 2 column 0:0: expected "2023-10-23T13:52:55.015787878Z" but was "2023-10-23T13:55:01.543123456Z"
row 3 column 0:a list containing
row 3 column 0:0: expected "2023-10-23T13:51:54.732102837Z" but was "2023-10-23T13:53:55.832987654Z"
row 4 column 0:a list containing
row 4 column 0:0: expected "2023-10-23T13:33:34.937193Z" but was "2023-10-23T13:52:55.015787878Z"
row 5 column 0:a list containing
row 5 column 0:0: expected "2023-10-23T12:27:28.948Z" but was "2023-10-23T13:51:54.732102837Z"
row 6 column 0:a list containing
row 6 column 0:0: expected "2023-10-23T12:15:03.360103847Z" but was "2023-10-23T13:33:34.937193Z"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 3 failures in test test {date_nanos.Date_nanos to date nanos, index version SYNC} (60.0% fail rate in 5 executions)
- [main] 2 failures in step 8.16.0_bwc-snapshots (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -302,9 +302,6 @@ tests:
 - class: org.elasticsearch.integration.KibanaUserRoleIntegTests
   method: testSearchAndMSearch
   issue: https://github.com/elastic/elasticsearch/issues/113593
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {date_nanos.Date_nanos to date nanos, index version SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/113632
 - class: org.elasticsearch.xpack.transform.integration.TransformIT
   method: testStopWaitForCheckpoint
   issue: https://github.com/elastic/elasticsearch/issues/106113

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date_nanos.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date_nanos.csv-spec
@@ -204,7 +204,7 @@ null
 date nanos to long, index version
 required_capability: to_date_nanos
 
-FROM date_nanos | WHERE millis > "2020-02-02" | EVAL l = TO_LONG(nanos) | KEEP l;
+FROM date_nanos | WHERE millis > "2020-02-02" | EVAL l = TO_LONG(nanos) | SORT nanos DESC | KEEP l;
 
 l:long
 1698069301543123456
@@ -219,7 +219,7 @@ l:long
 long to date nanos, index version
 required_capability: to_date_nanos
 
-FROM date_nanos | WHERE millis > "2020-02-02" | EVAL d = TO_DATE_NANOS(num) | KEEP d;
+FROM date_nanos | WHERE millis > "2020-02-02" | EVAL d = TO_DATE_NANOS(num) | SORT nanos DESC | KEEP d;
 
 d:date_nanos
 2023-10-23T13:55:01.543123456Z
@@ -234,7 +234,7 @@ d:date_nanos
 date_nanos to date nanos, index version
 required_capability: to_date_nanos
 
-FROM date_nanos | WHERE millis > "2020-02-02" | EVAL d = TO_DATE_NANOS(nanos) | KEEP d;
+FROM date_nanos | WHERE millis > "2020-02-02" | EVAL d = TO_DATE_NANOS(nanos) | SORT nanos DESC | KEEP d;
 
 d:date_nanos
 2023-10-23T13:55:01.543123456Z

